Major configuration and folder structure rework

filipowm opened the pull request.

filipowm committed:
fix(config): :bug: use env variable for URL or default to localhost

filipowm committed:
chore: add new exclusions to gitignore and dockerignore

filipowm committed:
chore: :art: move docker-related code to dedicated directory

filipowm committed:
chore: :art: move site configuration to dedicated directory

ghost commented:
### Bug Fixes

*  **config:** :bug: use env variable for URL or default to localhost ([a55af6b](https://github.com/filipowm/gatsby-gitbook-starter/commit/a55af6b07678743c355e748f241fc05c2192e5a1))
*  **sidebar:** :bug: fix issue when multiple items are added to none group ([a6a7aaa](https://github.com/filipowm/gatsby-gitbook-starter/commit/a6a7aaa80f7965e683a63d39bff76b061ead36de))
*  **config:** watching changes in config files with reload in develop mode ([a4d657c](https://github.com/filipowm/gatsby-gitbook-starter/commit/a4d657c631ed51c6061fb2985ccf7ad743db4371))
* temporarily do not require config in algolia ([90b3f29](https://github.com/filipowm/gatsby-gitbook-starter/commit/90b3f29093ff4abd9d4b5e246c45c1e3081cd8a8))


### Chore

* add new exclusions to gitignore and dockerignore ([88f4aa0](https://github.com/filipowm/gatsby-gitbook-starter/commit/88f4aa071fab32d18b9aee5a000d2c2688b18c0c))
* :art: move docker-related code to dedicated directory ([4b82bc7](https://github.com/filipowm/gatsby-gitbook-starter/commit/4b82bc722dad702ae81d922d24e253e623e09468))
* :art: move site configuration to dedicated directory ([361646b](https://github.com/filipowm/gatsby-gitbook-starter/commit/361646b1fe7508fdafe7f984b3da40cea3b9cb40))


### Code Refactoring

* :recycle: use config-reader for loading site configuration ([fb9c766](https://github.com/filipowm/gatsby-gitbook-starter/commit/fb9c7662017b62e5dfeaf44edfa5f0f916caf248))
*  **config:** refactor: switch from JSON config to YAML ([6ee31ea](https://github.com/filipowm/gatsby-gitbook-starter/commit/6ee31ea2fc8fb3b3212a956d708474035476f88f))


### Features

*  **config:** :sparkles: add support for using yaml configuration ([0350689](https://github.com/filipowm/gatsby-gitbook-starter/commit/035068934253d89ba081f965cd23d70554ce35cc))
*  **config:** :sparkles: support for setting up config using env vars ([bb4f716](https://github.com/filipowm/gatsby-gitbook-starter/commit/bb4f71617d15e931c009aa713104ed0e9b2fa7ac))
*  **config:** use generate config.js imported by frontend code ([dbc32b5](https://github.com/filipowm/gatsby-gitbook-starter/commit/dbc32b5d6779c21f01d7c32fd7b06dc39b479e03))


### Contributors

@filipowm

filipowm committed:
refactor: :recycle: use config-reader for loading site configuration

filipowm committed:
feat(config): :sparkles: add support for using yaml configuration

filipowm committed:
feat(config): :sparkles: support for setting up config using env vars

filipowm committed:
refactor(config): refactor: switch from JSON config to YAML

filipowm committed:
feat(config): use generate config.js imported by frontend code

filipowm committed:
fix(sidebar): :bug: fix issue when multiple items are added to none group

create default root group, where any unassigned pages are added to. This fixes the issue, where if added more than 1 page to "none", then only one was rendered. It was because every time "root" group was being overriden.

filipowm committed:
fix(config): watching changes in config files with reload in develop mode

filipowm committed:
fix: temporarily do not require config in algolia
